### PR TITLE
Iterate node lists with foreach

### PR DESCRIPTION
**Pull request by jtojnar**
`DOMNodeList` implements `Traversable`.

There are some `for` loops left but we cannot simply replace those: PHP follows the DOM specification, which requires that `NodeList` objects in the DOM are live. As a result, any operation that removes a node list member node from its parent (such as `removeChild`, `replaceChild` or `appendChild`) will cause the next node in the iterator to be skipped.

We could work around that by converting those node lists to static arrays using `iterator_to_array` but not sure if it is worth it.

Also add few cleanups I noticed.

### CODE REVIEW

File: src/Readability.php [hunk 1]
```diff
@@ -1076,11 +1079,6 @@ protected function grabArticle(?\DOMElement $page = null)
                 }
             }
 
-            $candidates = $xpath->query('.//*[not(self::body) and (@class or @id or @style) and ((number(@readability) < 40) or not(@readability))]', $page->documentElement);
-
-            for ($c = $candidates->length - 1; $c >= 0; --$c) {
-                $node = $candidates->item($c);
-            }
```
Comment (j0k3r) [inline src/Readability.php:1083]: Why did you remove that part?

**Review by j0k3r — COMMENTED** [review]
I still found it more readable to read `foreach` than `for` loop.
So it's ok for me

File: src/Readability.php (hunk 1 above)
Comment (jtojnar) [reply src/Readability.php:1083]: https://github.com/jtojnar/php-readability/commit/b580cf216d9001f82c866bb9a6c8bcad1cc862d8 moved the code that used the `$node` variable so now it is not used as far as I can tell.

**Comment by jtojnar** [comment]
I have rebased this, deciding to skip the reverse for loops for now – the naïve port to `iterator_to_array` does not really work (see #91, possibly due to nested nodes being matched by xpath) . Maybe it would work with `array_reverse` but I will need too give the invariants deeper thought.

**Review by j0k3r — APPROVED** [review]
